src/lib/storage/disk_size.rb

```diff
@@ -130,6 +130,7 @@ def unit_multiplier(unit)
       #
 
       def +(other)
+        return DiskSize.unlimited if any_operand_unlimited?(other)
```

Comment:
Um, I didn't know that as well, but when I saw your approach I could not resist to type "Ruby infinity" into Google... :wink: 

If it works for you then you can keep it, I was just looking for some more standard way...